### ISSUE TITLE
cronjob reapply fails with kubectl v 1.5.1 and v 1.5.2 against a v 1.5.1 cluster

### ISSUE THREAD

**Issue by das-vinculum** [kubernetes/kubernetes#39872]
**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
* kubectl, apply cronjob
* cronjob, reapply

---

**BUG REPORT**:
**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T07:30:54Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1+coreos.0", GitCommit:"cc65f5321f9230bf9a3fa171155c1213d6e3480e", GitTreeState:"clean", BuildDate:"2016-12-14T04:08:28Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: AWS using CoreOS
- **OS** (e.g. from /etc/os-release): Mac OS 10.12.2
- **Kernel** (e.g. `uname -a`):Darwin Florians-MacBook-Pro.local 16.3.0 Darwin Kernel Version 16.3.0: Thu Nov 17 20:23:58 PST 2016; root:xnu-3789.31.2~1/RELEASE_X86_64 x86_64
- **Install tools**:
- **Others**:

**What happened**:
I applied a cronjob definition using `kubectl apply -f cronjob.yaml` this was successful and the job actually got started. But when changing the cronjob, e.g. the time of the run, another `kubectl apply -f cronjob.yaml` fails with:
```
error: error when applying patch:

to:
&{0xc42056f680 0xc4202e7730 default mongodb-backup cronjob.yaml &CronJob{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:mongodb-backup,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{kubectl.kubernetes.io/last-applied-configuration: ,},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:CronJobSpec{Schedule:* 14 * * *,StartingDeadlineSeconds:nil,ConcurrencyPolicy:,Suspend:nil,JobTemplate:JobTemplateSpec{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:JobSpec{Parallelism:nil,Completions:nil,ActiveDeadlineSeconds:nil,Selector:nil,ManualSelector:nil,Template:k8s_io_kubernetes_pkg_api_v1.PodTemplateSpec{ObjectMeta:ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:PodSpec{Volumes:[{mongodb-backup {nil nil nil nil nil nil nil nil nil PersistentVolumeClaimVolumeSource{ClaimName:mongodb-backup-volume,ReadOnly:false,} nil nil nil nil nil nil nil nil nil nil nil nil nil}}],Containers:[{mongodb-backup 074509403805.dkr.ecr.eu-west-1.amazonaws.com/rebuy-mongodb-backup:latest [] []  [] [] {map[] map[]} [{mongodb-backup false /mongodb-backup }] nil nil nil  IfNotPresent nil false false false}],RestartPolicy:OnFailure,TerminationGracePeriodSeconds:nil,ActiveDeadlineSeconds:nil,DNSPolicy:,NodeSelector:map[string]string{},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:nil,ImagePullSecrets:[],Hostname:,Subdomain:,},},},},},Status:CronJobStatus{Active:[],LastScheduleTime:<nil>,},} &TypeMeta{Kind:,APIVersion:,} 10604 false}
for: "cronjob.yaml": error when serializing current configuration from:
&TypeMeta{Kind:,APIVersion:,}
for: "cronjob.yaml": no kind "CronJob" is registered for version "batch/v1"
```
Deleting and applying the same configuration was flawlessly possible.

**What you expected to happen**:
I expected to see that the call `kubectl apply -f cronjob.yaml` would return `cronjob "name" configured` and the changes get appied.

**How to reproduce it** (as minimally and precisely as possible):
* Take the example cron job definition from: https://kubernetes.io/docs/user-guide/cron-jobs/
* Apply it to the cluster
* Change for example the times for execution
* Apply it to the cluster

**Anything else do we need to know**:
We are running a custom cluster, but it already runs productive code to our satisfaction, so we do not expect a configuration issue.

Feel free to approach me for any open questions.

**Comment by diclophis** [comment]
I am running into this same issue for a simple CronJob resource.

**Comment by jasonswat** [comment]
I got that same error, but was able to use "replace" instead of "apply" which did what I was trying to do:

```
$ kubectl create -f snapshot-scheduledjob.yml
$ kubectl get cronjobs
NAME                  SCHEDULE         SUSPEND   ACTIVE    LAST-SCHEDULE
snapshot-job   5 0,8,16 * * *   False     0         <none>
```

I updated snapshot-scheduledjob.yml with a new schedule of 1,8,16

```
$ kubectl replace -f snapshot-scheduledjob.yml
$ kubectl get cronjobs
NAME                  SCHEDULE         SUSPEND   ACTIVE    LAST-SCHEDULE
snapshot-job   5 1,8,16 * * *   False     1         Thu, 09 Feb 2017 08:05:00 -0800
```

**Comment by bjoernhaeuser** [comment]
This issue can be closed, its not happening with 1.6 anymore.
